addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```
@@ -279,6 +280,15 @@ export class Wysiwyg extends Component {
                 this.env.onWysiwygReset && this.env.onWysiwygReset();
             }
         });
+        this.isPortalUser = false;
+        this.userService.hasGroup('base.group_portal')
+            .then(function(isPortal) {
+                this.isPortalUser = isPortal;
+            }.bind(this))
+            .catch(function(error) {
+                //  Might assume non-portal if in doubt.
+                this.isPortalUser = false;
+            }.bind(this));
```

Turns out that `onWillUpdateProps` can be an `async` function so you can add `async` on line 251 then simply write:
```
this.isPortalUser = await this.userService.hasGroup('base.group_portal')
```

---

Hello guys :wave:,

If the admin changes the access rights or the record rules in the settings and grants the portal users the rights to create attachments, wouldn't it be a problem to hardcode the group here?

What if we just do the call and we let the request fail? Then, in our `try/catch`, we determine the source of the error and we decide to either silent the error or re-throw it when it's necessary.

By doing this, we avoid making an extra call to check the access rights and we ensure that the code is independent of the access rights set on the database. **What do you think?**

---

That's an interesting proposition that sits in the middle of what @elhayyany originally did which was catching every errors and what we suggested him to do which was to check access rights before uploading. Could you try that @elhayyany please?

---

I'll update the commit asap

---

I updated it, what I did is checking if the current user a portal user,  Silently keep base64. 
I want to check the error but the error is general, so I checked if it is a portal user instead.

---

Hello @elhayyany, I think you are missing the point. I think what @julien-banken wanted specifically was to avoid calling `hasGroup` unless there was an error. In your current code on the PR you always call `hasGroup` and only read its value when there is an error.